src/codemagic/tools/app_store_connect/mixins/signing_file_saver_mixin.py

```diff
@@ -0,0 +1,82 @@
+from __future__ import annotations
+
+import pathlib
+import re
+import tempfile
+from typing import List
+from typing import Optional
+from typing import Sequence
+
+from codemagic.apple.resources import Profile
+from codemagic.apple.resources import SigningCertificate
+from codemagic.models import Certificate
+from codemagic.models import PrivateKey
+
+from ..errors import AppStoreConnectError
+from ..resource_printer import ResourcePrinter
+
+
+class SigningFileSaverMixin:
+    certificates_directory: pathlib.Path
+    printer: ResourcePrinter
+    profiles_directory: pathlib.Path
+
+    @classmethod
+    def _get_unique_path(cls, file_name: str, destination: pathlib.Path) -> pathlib.Path:
```

Comment:
Originally defined in https://github.com/codemagic-ci-cd/cli-tools/blob/3670fa624997f17cb437bc88198f7bdee3594f35/src/codemagic/tools/app_store_connect.py#L1029-L1042